packages/makeswift/src/integrate-next-app.ts

```diff
@@ -214,16 +216,26 @@ export default function Page({ snapshot }: Props) {
   const catchAllRoute = generateCatchAllRoute(isTypeScript)
   const extension = isTypeScript ? 'ts' : 'js'
 
-  // @todo: need to detect if a catch all route already exists
-  const catchAllRouteFilename =
-    fs.existsSync(path.join(pagesFolder, `index.${extension}`)) ||
-    fs.existsSync(path.join(pagesFolder, `index.${extension}x`))
-      ? `[...path].${extension}x`
-      : `[[...path]].${extension}x`
-
   // catch-all-route does not exist
-  if (glob.sync(path.join(pagesFolder, `\\[*\\].${extension}*`)).length === 0) {
-    fs.writeFileSync(path.join(temporaryPagesFolder, catchAllRouteFilename), catchAllRoute)
+  if (glob.sync(path.join(pagesFolder.absolute, `\\[*\\].${extension}*`)).length === 0) {
+    const useOptionalCatchAllRoute =
+      !fs.existsSync(path.join(pagesFolder.absolute, `index.${extension}`)) &&
+      !fs.existsSync(path.join(pagesFolder.absolute, `index.${extension}x`))
+
+    let catchAllRouteFilename
+    if (useOptionalCatchAllRoute) {
+      catchAllRouteFilename = `[[...path]].${extension}x`
+    } else {
+      catchAllRouteFilename = `[...path].${extension}x`
+
+      console.log(
```

Comment:
We might want to add a URL here to the Next.js docs on dynamic routes. This is an advanced feature and I guess most people won't know what a normal vs non-normal catch-all route is